polylaue/ui/regions_navigator/dialog.py

```diff
@@ -0,0 +1,399 @@
+
+        self.add_button = QPushButton(START_INTERACTIVE_MSG, self)
+        self.show_button = QPushButton(HIDE_MSG, self)
+        self.remove_button = QPushButton(REMOVE_MSG, self)
+        self.display_button = QPushButton(DISPLAY_MSG, self)
+        self.remove_button.setEnabled(False)
+        self.display_button.setEnabled(False)
+        add_remove_layout.addWidget(self.add_button)
+        add_remove_layout.addWidget(self.show_button)
+        add_remove_layout.addWidget(self.remove_button)
+        add_remove_layout.addWidget(self.display_button)
+
+        self.resize(600, 300)
+
+        self.update_add_button()
+
+        self.setup_connections()
+
+    def setup_connections(self):
+        self.rejected.connect(self.on_close)
+        self.accepted.connect(self.on_close)
+        self.add_button.clicked.connect(self.on_add_clicked)
+        self.show_button.clicked.connect(self.on_show_clicked)
+        self.remove_button.clicked.connect(self.on_remove_clicked)
+        self.display_button.clicked.connect(self.on_display_clicked)
+        self.roi_items_manager.sigActiveROIChanged.connect(
+            self.on_active_roi_changed
+        )
+        self.roi_items_manager.sigROIModified.connect(self.sigRoiModified)
+        self.view.selectionModel().selectionChanged.connect(
+            self.on_selection_changed
+        )
+
+    def on_close(self):
+        # Disable interactive add when the dialog is closed
+        self.add_status = True
+        self.on_add_clicked()
+
+    def update_add_button(self):
+        if self.add_status:
+            msg = STOP_INTERACTIVE_MSG
+            color = Qt.red
+        else:
+            msg = START_INTERACTIVE_MSG
+            color = Qt.green
+
+        pal = self.add_button.palette()
+        pal.setColor(QPalette.Button, QColor(color))
+        self.add_button.setAutoFillBackground(True)
+        self.add_button.setPalette(pal)
+        self.add_button.setText(msg)
+        self.add_button.update()
+
+    def on_add_clicked(self):
+        self.add_status = not self.add_status
+
+        self.update_add_button()
+
+        if self.add_status:
+            self.roi_items_manager.enable_interactive_add()
+        else:
+            self.roi_items_manager.disable_interactive_add()
+
+    def on_show_clicked(self):
+        self.show_status = not self.show_status
+
+        if self.show_status:
+            self.show_button.setText(HIDE_MSG)
+        else:
+            self.show_button.setText(SHOW_MSG)
+
+        self.roi_items_manager.show_items(self.show_status)
+
+    def on_remove_clicked(self):
+        if self.selected_roi_id is None:
+            return
+
+        id = self.selected_roi_id
+        row = self.roi_manager.id_to_index(id)
+
+        self.model.beginRemoveRows(QModelIndex(), row, row)
+        self.roi_manager.remove_roi(id)
+        self.roi_items_manager.remove_roi_item(id)
+        self.model.endRemoveRows()
```

Comment:
I personally think it's better for any functions like `beginRemoveRows()` and `endRemoveRows()` to be called within the model class itself. So that might mean adding an extra function to the model like `self.model.remove_row(id)`, for instance, that takes care of these things internally.

Reply:
Agreed, created 3 methods on the model which are invoked when adding/removing/modifying rows